### ISSUE TITLE
Cannot regenerate parser from metaparser.gram, on Windows.

### ISSUE THREAD

**Issue by mrolle45** [python/cpython#101836]
# Bug report

I have been playing with the metaparser.gram file.  On linux, the makefile will regenerate grammar_parser.py.  But in PCBuild\build.bat, there's no way to do that.  build --regen only regenerates parser.c from python.gram.

As it is, I have to run the pegen module manually.

There should be additional lines in regen.targets to call pegen a second time with `python` instead of `c`, and the requisite input and output files.  Also, add these files to _PegenSources and _PegenOutputs.

- CPython versions tested on: 3.11.0
- Operating system and architecture: Windows 10

**Comment by mrolle45** [comment]
I have code for the fix on [GitHub](https://github.com/mrolle45/cpython/tree/bug-101836).
It adds the needed command to PCBuild\regen.targets.
The branch is based on the 3.11 branch.

**Comment by arhadthedev** [comment]
@pablogsal (as a parser expert)

**Comment by mrolle45** [comment]
My code is wrong, as I discovered.  When I did the build.bat --regen, after making a change to the metaparser.gram file, it was importing the grammar_parser.py file, which is the file it was trying to regenerate.
It does need grammar_parser to regenerate the c file from Python.gram, so the batch command to regenerate grammar_parser should be _before_, not after, the command to regenerate parser.c.
However, after making a new grammar_parser from a changed metagrammar. I got an error regenerating grammar_parser, because it was using an older version of that file.

I'll update my fork of cpython after I figure out how to deal with this import of out of date grammar_parser module.

**Comment by pablogsal** [comment]
CC @tonybaloney

**Comment by mrolle45** [comment]
Revised regen.targets is now [posted](https://github.com/mrolle45/cpython/tree/bug-101836).

I don't know yet about doing regen after changing metagrammar.  Stay tuned...

**Comment by tonybaloney** [comment]
Thanks, the steps should follow the sequence in the makefile 
https://github.com/mrolle45/cpython/blob/bug-101836/Makefile.pre.in#L1288-L1303

**Comment by mrolle45** [comment]
There is still a problem regenerating grammar_parser.  I made a change to metagrammar which resulted in a grammar_parser which failed, due to an undefined name.
I tried correcting the problem in a new metagrammar, but the regen was running the current grammar_parser to parse the new metagrammar, and so it still failed.

I don't know yet how to resolve this.  I think that pegen -python should check the result file by importing and running it against the input, and if this fails, then don't write the new output file.  Using the Makefile, it could just raise an exception and fail the step, and then the Makefile would skip the command to update the output file.  On Windows, the regen -python should fail, and the target rule would abort further steps.

Since this seems to be independent of the makefile or PCBuild, I'll close this issue and file a new issue on pegen.py.
